main/db.c

```diff
@@ -269,34 +255,11 @@ static int db_create_astdb(void)
 
 static int db_open(void)
 {
-	char *dbname;
-	struct stat dont_care;
+	char dbname[strlen(ast_config_AST_DB) + sizeof(".sqlite3")];
 
-	if (!(dbname = ast_alloca(strlen(ast_config_AST_DB) + sizeof(".sqlite3")))) {
-		return -1;
-	}
 	strcpy(dbname, ast_config_AST_DB);
 	strcat(dbname, ".sqlite3");
 
-	if (stat(dbname, &dont_care) && !stat(ast_config_AST_DB, &dont_care)) {
-		if (convert_bdb_to_sqlite3()) {
```

Comment:
For the off chance that someone is doing an ancient upgrade to new, how about just having an ERROR message stating that they need to use an earlier version of Asterisk to convert their database to sqlite3 if desired?